### PR TITLE
Fix Issue 21096 - [ICE] Segmentation fault in dmd.hdrgen.sizeToBuffer at dmd/hdrgen.d:3153

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: src/dmd/parse.d [hunk 1]
```diff
@@ -3958,6 +3958,12 @@ final class Parser(AST) : Lexer
                     //printf("it's type[expression]\n");
                     inBrackets++;
                     AST.Expression e = parseAssignExp(); // [ expression ]
+                    if (!e)
```
Comment (RazvanN7) [inline src/dmd/parse.d:3961]: If what was between the brackets has failed to parse, `e` is going to be null causing the segfault.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
```
~> ./build.sh
--
  | @@@@@ WARNING @@@@@
  | @ This script is DEPRECATED. Use build.d directly instead @
  | @@@@@@@@@@@@@@@@@@@@
  | ../dmd/generated/linux/release/64/../../../../../phobos/generated/linux/release/64/libphobos2.a(bitmanip_148_5db.o):std/bitmanip.d:function _D3std8bitmanip__T10swapEndianTlZQpFNaNbNiNfxlZl: error: undefined reference to '_D4core5bitop5bswapFNaNbNiNfmZm'
```
Not sure why buildkite is doing that.

**Comment by RazvanN7** [comment]
I'm fairly sure the fail is unrelated.

**Review by ghost — CHANGES_REQUESTED** [review]
Semaphore CI in the services is an indication that you must pull stable and rebase your PR on it.
 Everyting should be green after that.

**Comment by RazvanN7** [comment]
Semaphore is passing. Do you mean buildkite?

**Comment by thewilsonator** [comment]
Thats what I copied the error message from.

**Comment by NilsLankila** [comment]
My mistake, I thought that we dont use Semaphore anymore. Was your local stable up to date ?

**Comment by RazvanN7** [comment]
Yes.

**Review by ghost — APPROVED** [review]

**Comment by NilsLankila** [comment]
That's strange,  I never see Semaphore in the my own not so old merged PR.

**Comment by Geod24** [comment]
> That's strange, I never see Semaphore in the my own not so old merged PR.

That's because the token got invalidated for some reason so it was disabled. I just fixed it today.

**Comment by ibuclaw** [comment]
> My mistake, I thought that we dont use Semaphore anymore. Was your local stable up to date ?

Still need semaphore to check gdc and ldc can bootstrap master/stable. :-)

**Pull request merged**